refiner-app/src/socket.rs

```diff
@@ -0,0 +1,295 @@
+use std::path::Path;
+
+use actix_rt::net::{UnixListener, UnixStream};
+use actix_rt::Arbiter;
```

Comment:
Personally I would use tokio directly for this instead of going through actix. I know `main.rs` currently uses actix as its runtime, but I would change that to use tokio directly as well.

Reply:
That makes sense. I used the actix versions mostly to keep it consistent with previous code but they're are re-exports anyway. Switched actix to tokio in 7af7255.

Should we handle shutdown properly by passing a signal and then waiting on tasks to complete? The socket file remains if not deleted explicitly.
Also added the socket server in 4e3b011 to main.

Reply:
Nice, looks good. Yeah, it is probably a good idea to try to do a graceful shutdown. Though it does increase the complexity of the code a bit.

Reply:
Added stop signals in 3d66793